### PR TITLE
issue-756: Highlight selected chat in left chat list panel

### PR DESCRIPTION
**Pull request by zainashraf1562**
Issue: https://github.com/monal-im/Monal/issues/756

Solved Video Clip:
https://github.com/monal-im/Monal/assets/155921778/61aca67a-5b1f-498e-93cf-3143fbbeb89b

### CODE REVIEW

File: Monal/Classes/ActiveChatsViewController.m [hunk 1]
```diff
@@ -42,6 +42,7 @@ @interface ActiveChatsViewController() {
 }
 @property (atomic, strong) NSMutableArray* unpinnedContacts;
 @property (atomic, strong) NSMutableArray* pinnedContacts;
+@property (nonatomic, strong) MLContact *selectedChat;
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ActiveChatsViewController.m]: don't do this, but use `[MLNotificationManager sharedInstance].currentContact` (see line 575 of `ActiveChatsViewController.m` for example)

Comment (tmolitor-stud-tu) [reply Monal/Classes/ActiveChatsViewController.m]: Also: Please stick to our coding style guideline: https://github.com/monal-im/Monal/wiki/Code-style/

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]

**Comment by zainashraf1562** [comment]
Updated clip showing pinned chat scenario: 
https://github.com/monal-im/Monal/assets/155921778/92abc9e2-9d09-4a39-908f-4ed153575de2

Recommendation for pinned chat: Instead of using a color just include a pinned Icon into the Chat Cell. If visibility is in question you can just make the icon bigger or a brighter color to make it pop.(issue #755)

Also a more intuitive way to add a chat would be to double tap it. Right now double tapping a cell is causing it to go into the first position not sure if thats on purpose or a failed attempt at pinning a chat. (Most likely reason for incorrect chat order issue #961)

FYI If you want me to work on any of them let me know. As a gsoc applicant i already have 2 short teaser tasks assigned not sure if i could to do more as it may hinder with other applicants process.

**Comment by tmolitor-stud-tu** [comment]
> Recommendation for pinned chat: Instead of using a color just include a pinned Icon into the Chat Cell. If visibility is in question you can just make the icon bigger or a brighter color to make it pop.(issue https://github.com/monal-im/Monal/issues/755)

A Pin icon sounds good please implement that (without a solution to pinning, the highlighting of the currently visible chat is incomplete because it won't be immediately obvious if a pinned chat is currently selected or vice versa if a currently selected chat is pinned).
Pinning is currently made clear by the color of the chat list entry btw.

> Also a more intuitive way to add a chat would be to double tap it. Right now double tapping a cell is causing it to go into the first position not sure if thats on purpose or a failed attempt at pinning a chat. (Most likely reason for incorrect chat order issue https://github.com/monal-im/Monal/issues/961)

No, tapping onto a chat should only open it. it's a bug that the chat list gets sorted by this action. pinning should only be done using the appropriate button in the contact details
I don't think pinning is an action often enough executed to make a shortcut like double tapping necessary.

**Comment by zainashraf1562** [comment]
Updated clip with pinned Icon:

https://github.com/monal-im/Monal/assets/155921778/f56736a6-ff1d-45ea-87fb-85c248189039

**Comment by tmolitor-stud-tu** [comment]
We are using a rebase-based workflow rather than a merge-based and sometimes force-push something into one of the top commits.

Please rebase this PR upon current develop and remove all commits not being yours by calling: `git rebase -i upstream/develop` (assuming you added the remote named "upstream" pointing to our github repo).

File: Monal/Classes/MLContactCell.m [hunk 2]
```diff
@@ -200,11 +200,11 @@ -(void) setCount:(long)count
 -(void) setPinned:(BOOL) pinned
 {
     self.isPinned = pinned;
-
-    if(pinned) {
-        self.backgroundColor = [UIColor colorNamed:@"activeChatsPinnedColor"];
+    
+    if (pinned) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLContactCell.m]: objc style: `{` into next line; no space between `if` and `(`, see https://github.com/monal-im/Monal/wiki/Code-style/

**Comment by zainashraf1562** [comment]
@tmolitor-stud-tu rebased.
Screenshot 2024-03-23 at 3 18 22 PM

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
Ah well, you did not follow our coding style guidlines in this PR, please follow our guidlines exactly: https://github.com/monal-im/Monal/wiki/Code-style/

File: Monal/Classes/MLContactCell.m (hunk 2 above)
Comment (tmolitor-stud-tu) [reply Monal/Classes/MLContactCell.m]: here it is, please fix!

**Comment by tmolitor-stud-tu** [comment]
Please also squash all commits into one and give it a commit message looking like "<your text>, fixes #756", so that the linked issue will automatically be closed...

**Comment by tmolitor-stud-tu** [comment]
Please change the commit message to "Highlight selected chat, **fixes #756**"

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
You forgot to change the commit message to something useful:
Please change the commit message to "Highlight selected chat, **fixes #756**"

File: Monal/Classes/ActiveChatsViewController.m [hunk 3]
```diff
@@ -723,6 +723,16 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 
     [cell initCell:chatContact withLastMessage:messageRow];
 
+    cell.selectionStyle = UITableViewCellSelectionStyleNone;
+    
+    // Highlight the selected chat
+    if ([MLNotificationManager sharedInstance].currentContact != nil && [chatContact isEqual:[MLNotificationManager sharedInstance].currentContact]) {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ActiveChatsViewController.m:729]: You still did not follow our coding guidlines!
I will merge this nevertheless, but I don't want to have myself repeat here in the future

**Review by tmolitor-stud-tu — APPROVED** [review]
You still did not follow our coding guidlines!
I will merge this nevertheless, but I don't want to have myself repeat here in the future

**Pull request merged**